src/server/active_tran_server.cpp

```diff
@@ -190,22 +190,24 @@ active_tran_server::connect_to_page_server (const cubcomm::node &node, const cha
       return ER_NET_PAGESERVER_CONNECTION;
     }
 
-  er_log_debug (ARG_FILE_LINE, "Successfully connected to the page server. Channel id: %s.\n",
-		srv_chn.get_channel_id ().c_str ());
-
-  m_ps_conn.reset (new page_server_conn (std::move (srv_chn)));
-  m_ps_conn->register_request_handler (ps_to_ats_request::SEND_SAVED_LSA,
-				       std::bind (&active_tran_server::receive_saved_lsa, std::ref (*this),
-					   std::placeholders::_1));
-  m_ps_conn->register_request_handler (ps_to_ats_request::SEND_LOG_PAGE,
-				       std::bind (&active_tran_server::receive_log_page, std::ref (*this), std::placeholders::_1));
-  m_ps_conn->register_request_handler (ps_to_ats_request::SEND_DATA_PAGE,
-				       std::bind (&active_tran_server::receive_data_page, std::ref (*this), std::placeholders::_1));
-  m_ps_conn->start_thread ();
-
-  m_ps_request_queue.reset (new page_server_request_queue (*m_ps_conn));
-  m_ps_request_autosend.reset (new page_server_request_autosend (*m_ps_request_queue));
-  m_ps_request_autosend->start_thread ();
+  const std::string channel_id { srv_chn.get_channel_id () };
+  assert (m_ps == nullptr);
+  m_ps.reset (new ps_t (std::move (srv_chn),
+  {
+    {
+      ps_to_ats_request::SEND_SAVED_LSA,
+      std::bind (&active_tran_server::receive_saved_lsa, std::ref (*this), std::placeholders::_1)
+    },
+    {
+      ps_to_ats_request::SEND_LOG_PAGE,
+      std::bind (&active_tran_server::receive_log_page, std::ref (*this), std::placeholders::_1)
+    },
+    {
+      ps_to_ats_request::SEND_DATA_PAGE,
+      std::bind (&active_tran_server::receive_data_page, std::ref (*this), std::placeholders::_1)
+    }
+  }));
+  er_log_debug (ARG_FILE_LINE, "ATS connected to PS. Channel id: %s.\n", channel_id.c_str ());
```

Comment:
Move er_log back before m_ps is created. The successful connection is confirmed by channel.send_int operation.

Reply:
Done